src/manufacturer/manufacturerDialog.component.tsx

```diff
@@ -222,53 +211,34 @@ function ManufacturerDialog(props: ManufacturerDialogProps) {
       const isURLUpdated =
         manufacturer.url !== selectedManufacturer.url &&
         manufacturer.url !== undefined;
-      const isBuildingNumberUpdated =
-        manufacturer.address?.building_number !==
-        selectedManufacturer.address.building_number;
-      const isStreetNameUpdated =
-        manufacturer.address?.street_name !==
-        selectedManufacturer.address.street_name;
+      const isAddressLineUpdated =
+        manufacturer.address?.address_line !==
+        selectedManufacturer.address.address_line;
       const isTownUpdated =
         manufacturer.address?.town !== selectedManufacturer.address.town;
       const isCountyUpdated =
         manufacturer.address?.county !== selectedManufacturer.address.county;
-      const ispostcodeUpdated =
+      const isPostcodeUpdated =
         manufacturer.address?.postcode !==
         selectedManufacturer.address.postcode;
+      const isCountryUpdated =
+        manufacturer.address?.country !== selectedManufacturer.address.country;
       const isTelephoneUpdated =
         manufacturer.telephone !== selectedManufacturer.telephone;
```

Comment:
I think these may need modifying - where they are optional, the types will always be different so the result will always be true e.g. `selectedManufacturer.telephone` can be None, but `manufacturer.telephone` would be "" if I added then deleted something again. It could be a case of adding `value || null` on the onChange methods so they are stored as the same types, which should hopefully also resolve them being added to the address data sent as currently changing the country on its own sends
```
{"address":{"address_line":"Address Line","town":null,"county":null,"country":"Country3","postcode":"Postcode"}}
```
But ideally it would just be
```
{"address":{"country":"Country3"}}
```